src/transport/CertificatePairingSession.h

```diff
@@ -0,0 +1,175 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the CHIP CASE Session object that provides
+ *      APIs for constructing secure session using certificate from device's
+ *      operational credentials.
+ */
+
+#pragma once
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <protocols/secure_channel/Constants.h>
+#include <support/Base64.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/PairingSessionDelegate.h>
+#include <transport/PeerConnectionState.h>
+#include <transport/SecureSession.h>
+#include <transport/raw/MessageHeader.h>
+#include <transport/raw/PeerAddress.h>
+
+namespace chip {
+
+using namespace Crypto;
+
+class DLL_EXPORT CertificatePairingSession
```

Comment:
Pairing is overall is counter-intuitive and non-spec-compliant naming. Both PASE and CASE are *Session Establishment* protocols. A better name would be `CertificateAuthenticatedSession`